### PR TITLE
Fix The application badge should be updated even if the device is off… #531

### PR DESCRIPTION
**Pull request by pedroGitt**
Fix the badge update when the device is offline
Update of the SDK lib with its last version

### CODE REVIEW

**Comment by ylecollen** [comment]
The invited rooms are not counted to update the badge (need to check if the dedicated rule is enabled / disabled)

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java [hunk 1]
```diff
@@ -774,6 +779,47 @@ private void markAllMessagesAsRead() {
...
+     * message of the room.
+     * @return true if operation was performed, false otherwise
+     */
+    private boolean markAllMessagesAsReadWhenOffline() {
+        boolean isOperationDone = false;
+
+        if(!Matrix.getInstance(this).isConnected()) {
+
+            ArrayList<MXSession> sessionsList = new ArrayList<>(Matrix.getMXSessions(this));
+            mReadReceiptSessionListIterator = sessionsList.iterator();
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: it seems being a copy & paste from the standard case.
It seems useless here.

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java [hunk 2]
```diff
@@ -774,6 +779,47 @@ private void markAllMessagesAsRead() {
...
+                for (MXSession session : sessionsList) {
+                    if (null != session) {
+                        // test if the session is still alive i.e the account has not been logged out
+                        ArrayList<Room> roomCompleteList = new ArrayList<>(session.getDataHandler().getStore().getRooms());
+
+                        if(null != roomCompleteList) {
+                            // for each room send the receipt for the latest message
+                            for (Room room : roomCompleteList) {
+                                room.sendReadReceipt(null);
+                                isOperationDone = true;
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: isOperationDone |= room.sendReadReceipt(null);
seems better : it is possible there is no update. sendReadReceipt returns true when there is an update.

File: vector/src/main/java/im/vector/activity/CommonActivityUtils.java [hunk 3]
```diff
@@ -1684,6 +1685,50 @@ public static int getBadgeCount() {
...
     }
 
+    /**
+     * Refresh the badge count when the device is offline.
+     * Notifications rooms are parsed to track the notification count value.
+     * @param aSession session value
+     * @param aActivity calling activity
+     */
+    public static void offlineRefreshBadgeUnreadCount(MXSession aSession, Activity aActivity) {
+        MXDataHandler dataHandler;
```
Comment (ylecollen) [inline vector/src/main/java/im/vector/activity/CommonActivityUtils.java]: i don't understand why an Activity is required here.
A context seems better.
For the linked f-droid issue, an activity won't be always available.

Comment (pedroGitt) [inline vector/src/main/java/im/vector/activity/CommonActivityUtils.java]: ACK

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java (hunk 2 above)
Comment (pedroGitt) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: ACK

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java (hunk 1 above)
Comment (pedroGitt) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: ACK

**Pull request merged**